docs/DeveloperGuide.md

```diff
@@ -1,25 +1,88 @@
 # Developer Guide
```

Comment:
Is it better to have more visuals like sequence diagrams for better illustration?